sparkmagic/sparkmagic/livyclientlib/livysession.py

```diff
@@ -252,12 +260,12 @@ def sleep(self):
         sleep(self._statement_sleep_seconds)
 
     def refresh_status(self):
-        status = self._http_client.get_session(self.id)[u'state']
+        response = self._http_client.get_session(self.id)
+        status = response[u'state']
+        log_array = response[u'log']
 
-        if status in constants.POSSIBLE_SESSION_STATUS:
-            self.status = status
-        else:
-            raise LivyUnexpectedStatusException(u"Status '{}' not supported by session.".format(status))
+        self.status = status
+        self._logs = "\n".join(log_array)
```

Comment:
The problem is we need it to be atomic. And the Yarn diagnostic log changes, it is not accumulative. So the get status call issued in Refresh_status and get_logs could have different states and different logs.
Current refresh_status does have side effect which is changing the state in the object. By changing the name to refresh_status_and_log, it is indicating what it is doing. If you are strict about the return type, we can change this function to not return anything and just refresh the internal variables and the caller can call LivySession.status to retrieve the value. To me that is no ideal either.